sw/device/lib/base/bitfield.c

```diff
@@ -6,8 +6,15 @@
 
 // `extern` declarations to give the inline functions in the
 // corresponding header a link location.
-extern uint32_t bitfield_set_field32(uint32_t bitfield,
-                                     bitfield_field32_t field);
 
-extern uint32_t bitfield_get_field32(uint32_t bitfield,
-                                     bitfield_field32_t field);
+extern uint32_t bitfield_field32_read(uint32_t bitfield,
+                                      bitfield_field32_t field);
+extern uint32_t bitfield_field32_write(uint32_t bitfield,
+                                       bitfield_field32_t field,
+                                       uint32_t value);
+
+extern bool bitfield_bit32_read(uint32_t bitfield,
+                                bitfield_bit32_index_t bit_index);
+extern uint32_t bitfield_bit32_write(uint32_t bitfield,
+                                     bitfield_bit32_index_t bit_index,
+                                     bool value);
```

Comment:
With bits and fields, I find it more descriptive to name them `set/get` than `read/write`. I am not against `read/write`, but I think it is more work (changing all the DIFs) for no gain.

Reply:
I have read it. It doesn't quite apply to single bit operations. Sure, maybe it makes it clearer in case of `set/clear_field` API, but I don't think there was confusion in the first place.

Reply:
I agree that this is clearer. I have only used clear/set by convention, but I feel like they're not good great names, especially at a software level where we don't distinguish the operations as strongly.

Reply:
Inspiration for the names came from the CSR modification instructions in RISC-V, which use the read/write/set/clear convention (providing all four operations)

Reply:
Fair enough, just my personal preference - anyway it looks good either way. I will resolve this thread.